assembl/views/api2/discussion.py

```diff
@@ -822,3 +827,78 @@ def test_results(request):
+def post_discussion(request):
+    from assembl.models import EmailAccount, User, LocalUserRole, Role, AbstractAgentAccount
+    ctx = request.context
+    json = request.json_body
+    user_id = authenticated_userid(request) or Everyone
+    permissions = get_permissions(user_id, None)
+    is_etalab_request = (request.matched_route and request.matched_route.name == 'etalab_discussions')
+    if is_etalab_request:
+        # The Etalab specification says that the API call representing the instance creation request must contain the following fields:
+        # - requestIdentifier
+        # - name: the title of the discussion (discussion.topic)
+        # - slug
+        # - adminName
+        # - adminEmail
+        default_view = 'etalab'
+        # Fake an APIv2 context
+        from ..traversal import Api2Context
+        ctx = Api2Context(ctx)
+        ctx = ClassContext(ctx, Discussion)
+        json['topic'] = json.get('name', json.get('slug', ''))
+    else:
+        default_view = 'default'
+    cls = ctx.get_class(json.get('@type', None))
+    typename = cls.external_typename()
+    # special case: find the user first.
+    creator_email = json.get("adminEmail", None)
+    db = Discussion.default_db
+    if creator_email:
+        account = db.query(AbstractAgentAccount).filter_by(
+            email=creator_email, verified=True).first()
+        if account:
+            user = account.profile
+        else:
+            user = User(name=json.get("adminName", None), verified=True)
+            account = EmailAccount(profile=user, email=creator_email, verified=True)
+            db.add(user)
+            db.flush()
+        json['creator'] = user.uri()
+    else:
+        user = None
+    try:
+        instances = ctx.create_object(typename, json, user_id)
+        discussion = instances[0]
+        # Hackish. Discussion API? Generic post-init method?
+        discussion.preferences.name = (
+            'discussion_' + json.get('slug', str(discussion.id)))
+        create_default_permissions(discussion)
+        if user is not None:
+            role = db.query(Role).filter_by(name=R_ADMINISTRATOR).first()
+            local_role = LocalUserRole(discussion=discussion, user=user, role=role)
+            instances.append(local_role)
+        discussion.invoke_callbacks_after_creation()
+    except ObjectNotUniqueError as e:
+        raise HTTPBadRequest(e)
+    except Exception as e:
+        raise HTTPServerError(e)
+    if instances:
+        first = instances[0]
+        db = first.db
+        for instance in instances:
+            db.add(instance)
+        db.flush()
+        view = request.GET.get('view', None) or default_view
+        uri = "/".join((API_ETALAB_DISCUSSIONS_PREFIX, str(first.id))) if is_etalab_request else None
+        return CreationResponse(first, user_id, permissions, view, uri=uri)
+
+
+def includeme(config):
+    # Make sure that the cornice view is registered
```

Comment:
Nothing here though? Should remove includeme since the cornice views are added by a `config.scan()` in the `assembl.views` module?